packages/utils/test/composeAbortSignals.test.ts

```diff
@@ -101,16 +100,14 @@ describe('composeAbortSignals', () => {
         })
     })
 
-    it('throws error if composing empty list', () => {
-        expect(() => {
-            composeAbortSignals()
-        }).toThrow('must provide at least one AbortSignal')
+    it('gives a pending signal for an empty list of signals', () => {
+        expect(composeAbortSignals().aborted).toBeFalse()
```

Comment:
I'm guessing accepting the empty list makes it easier to use:
```ts
const signals = [signal1, signal2, signal3, signal4, signal5].filter((v) => v !== undefined)

// old way:
let composedSignal
if (signals.length !== 0) {
    composedSignal = composeAbortSignals(signals)
}

// new way:
const composedSignal = composeAbortSignals(signals)
```

Originally when I wrote this, I wasn't really sure it made sense to have an abort signal that never aborts so I made it an error. But thinking about it now, I think there are cases where it makes sense such as `fetch`. The request will complete eventually anyway so passing it a signal that never aborts is not an issue.

I guess let's keep this new functionality, maybe add a new entry in PR description under "Changes" such as "composeAbortSignal now allows empty lists".

Reply:
Yeah, the "new way" could actually be written like so
```ts
const composedSignal = composeAbortSignals(
    signal1,
    signal2,
    signal3,
    signal4,
    signal5
)
```

No pre-filtering needed.